### PR TITLE
Getdata

### PR DESCRIPTION
**Pull request by tvu212**
All of the input data as well as the output data besides the spreadsheet now reads from the active sheet.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -21,15 +21,40 @@ var HEADERS = [
...
+var preAssignmentData = preAssignmentSheet.getDataRange().getValues();
+
 
 /**
  * Automatically runs when sheet is opened.
  */
 function onOpen() {
     SpreadsheetApp.getUi()
-        .createMenu("Great Explorations")
+        .createMenu("Great Explorations")spreadsheet
```
Comment (Chris-Pyles) [inline main.js]: "spreadsheet" looks like it shouldn't be here

File: main.js [hunk 2]
```diff
@@ -21,15 +21,40 @@ var HEADERS = [
...
 
-var OUTPUT_SHEET_ID = "13K10UA0ZNjCDGTbVbO104CdW97DJgm3MaK2TZpiRytw";
+// Formatting workshop variables
+
 var WORKSHOP_SHEET_ID = "1pZQWPV532JLWQuDLYiw4CdcvvBn8zoRQZ8lX2aaDzRc";
+var spreadsheet = SpreadsheetApp.getActiveSpreadsheet();
+
+
+// Response Data
+var responseSheet = spreadsheet.getSheets()[0];
```
Comment (Chris-Pyles) [inline main.js]: all instances of "spreadsheet.getSheets()[x] return undefined

File: main.js [hunk 3]
```diff
@@ -21,15 +21,40 @@ var HEADERS = [
...
+
+
+// Workshop Sheet
+var workshopSheet = SpreadsheetApp.openById(WORKSHOP_SHEET_ID);
+var workshopData = workshopSheet.getDataRange().getValues();
+
+
+// Output Sheet; Clears the sheet and reappends the header.
+var outputSheet = spreadsheet.getSheets()[1];
+outputSheet.clear();
```
Comment (Chris-Pyles) [inline main.js:46]: output sheet cannot be cleared since it's still undefined

**Review by Chris-Pyles — DISMISSED** [review]
Some of the sheet-reading doesn't seem to read the correct data. The only sheet currently defined correctly is the workshop sheet.

**Comment by tvu212** [comment]
It might not have been working since your personal test environment did not have have the other two sheets available which is why they were null or undeclared. I rewrote the code so now we are all pulling input and output to the same spreadsheet no matter what test environment we are in. So instead of using activesheet I call the ID number of the document.

**Comment by beszel** [comment]
@tvu212 Can you write out some steps for testing these changes?

File: main.js [hunk 4]
```diff
@@ -21,9 +21,35 @@ var HEADERS = [
...
+
 var WORKSHOP_SHEET_ID = "1pZQWPV532JLWQuDLYiw4CdcvvBn8zoRQZ8lX2aaDzRc";
+var RESPONSE_SHEET_ID = "1YcO_lYO1hp9j3fBxWm-4AF_AfYCPExeuVyaJzw0Yktk";
+
+var RESPONSE_SPREADSHEET = SpreadsheetApp.openById(RESPONSE_SHEET_ID);
+
+
+
+// Response Data
+var responseSheet = RESPONSE_SPREADSHEET.getSheets()[0];
```
Comment (beszel) [inline main.js]: Can you replace these numbers with variables? You could use an object to replace `[0]` with `[SHEET_INDICES.responses]`:
```
SHEET_INDICES = {
    responses: 0,
    output: 1
}
```

File: main.js [hunk 5]
```diff
@@ -21,9 +21,35 @@ var HEADERS = [
     "Session 3"
 ];
 
-var OUTPUT_SHEET_ID = "13K10UA0ZNjCDGTbVbO104CdW97DJgm3MaK2TZpiRytw";
+// Formatting workshop variables
+
 var WORKSHOP_SHEET_ID = "1pZQWPV532JLWQuDLYiw4CdcvvBn8zoRQZ8lX2aaDzRc";
+var RESPONSE_SHEET_ID = "1YcO_lYO1hp9j3fBxWm-4AF_AfYCPExeuVyaJzw0Yktk";
```
Comment (beszel) [inline main.js]: Can we change this variable name to reflect that the response data is only one sheet in this larger spreadsheet? We can use "SPREADSHEET" for both of these while we're at it to clarify that it's a spreadsheet ID.

File: main.js [hunk 6]
```diff
@@ -21,9 +21,35 @@ var HEADERS = [
...
+var RESPONSE_SPREADSHEET = SpreadsheetApp.openById(RESPONSE_SHEET_ID);
+
+
+
+// Response Data
+var responseSheet = RESPONSE_SPREADSHEET.getSheets()[0];
+var responseData = responseSheet.getDataRange().getValues();
 
+// Workshop Sheet
+var workshopSheet = SpreadsheetApp.openById(WORKSHOP_SHEET_ID);
```
Comment (beszel) [inline main.js]: Do these need to be defined globally, outside of a function? Can they be moved into a function and passed in as arguments where needed? If they need to be global, we can make them all capitals.

**Review by beszel — CHANGES_REQUESTED** [review]

**Comment by beszel** [comment]
Removed @Izayuh from reviewers because Chris and I are reviewing this one.

**Comment by beszel** [comment]
@tvu212 Can you address my requested changes?

**Comment by tvu212** [comment]
Yeah sounds good I'll get these done by tomorrow!

**Review by Chris-Pyles — APPROVED** [review]

File: main.js (hunk 6 above)
Comment (tvu212) [reply main.js]: Yes these should be global unless you want the function to be declared every time the workShop sheet is referenced. I'll change to make them all capitals

File: main.js (hunk 4 above)
Comment (tvu212) [reply main.js]: Done

File: main.js (hunk 5 above)
Comment (tvu212) [reply main.js]: Can you elaborate. I've made the changes I think you've wanted.

File: main.js [hunk 7]
```diff
@@ -21,28 +21,32 @@ var HEADERS = [
     "Session 3"
 ];
 
-// Formatting workshop variables
+//VARIABLES FOR RESPONSE SPREADSHEET INDICES
```
Comment (beszel) [inline main.js:24]: Why is this comment changed to capitals?

**Review by beszel — COMMENTED** [review]
This looks good, how can I test it?

File: main.js (hunk 5 above)
Comment (beszel) [reply main.js]: Looks good, that's all I had in mind.

**Pull request merged**